File: 백준/Silver/20153. 영웅이는 2의 거듭 제곱을 좋아해！ 영웅이는 2의 거듭 제곱을 좋아해！/영웅이는 2의 거듭 제곱을 좋아해！ 영웅이는 2의 거듭 제곱을 좋아해！.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int N;
    cin >> N;
    
    vector<int> nums(N);
    for (int i = 0; i < N; i++) {
        cin >> nums[i];
    }
    
    int xor_all = 0;
    for (int num : nums) {
        xor_all ^= num;
    }
    
    int result = 0;
    for (int num : nums) {
        result = max(result, xor_all ^ num);
    }
    result = max(result, xor_all); // 0과의 XOR은 원래 값과 같음
    
    cout << result << result << "\n";
    
    return 0;
}

File: 백준/Silver/20153. 영웅이는 2의 거듭 제곱을 좋아해！ 영웅이는 2의 거듭 제곱을 좋아해！/영웅이는 2의 거듭 제곱을 좋아해！ 영웅이는 2의 거듭 제곱을 좋아해！.py
import sys
input = sys.stdin.readline

N = int(input())
nums = list(map(int, input().split()))

xor_all = 0
for num in nums:
    xor_all ^= num

print(str(max(xor_all, max(xor_all ^ num for num in nums))) * 2)